plugins/org.polarsys.capella.cybersecurity.ui/src/org/polarsys/capella/cybersecurity/ui/properties/sections/SecurityNeedsSection.java

```diff
@@ -130,6 +131,9 @@ private EObject getSecurityNeedsObject(EObject parent) {
         ElementExtensionStorage<TrustBoundaryStorage> adapter = new ElementExtensionStorageImpl<>(
             (ExtensibleElement) parent, CybersecurityPackage.Literals.SECURITY_NEEDS);
         result = adapter.getExtension();
+        if (result instanceof SecurityNeeds) {
+          CybersecurityServices.setDefaultSecurityNeedsLevels((SecurityNeeds)result, parent);

```

Comment:
I think this might break what is written in :
https://github.com/eclipse/capella-cybersecurity/blob/c93b1427aac6ac2e8d86622b5deb89c58236c742/plugins/org.polarsys.capella.cybersecurity.ui/src/org/polarsys/capella/cybersecurity/ui/ElementExtensionStorageImpl.java#L27

the element SecurityNeed is not stored under the element unless if the user change one of its value. With this new method and since a transation.execute is called, when the user open the property view, it will create automatically the securityNeed no?

This method shall be put in https://github.com/eclipse/capella-cybersecurity/blob/c93b1427aac6ac2e8d86622b5deb89c58236c742/plugins/org.polarsys.capella.cybersecurity.ui/src/org/polarsys/capella/cybersecurity/ui/ElementExtensionStorageImpl.java#L51
instead in a transaction.
can you try to add a breakpoint in https://github.com/eclipse/capella-cybersecurity/blob/c93b1427aac6ac2e8d86622b5deb89c58236c742/plugins/org.polarsys.capella.cybersecurity.ui/src/org/polarsys/capella/cybersecurity/ui/ElementExtensionStorageImpl.java#L90 to see if the element is stored while call of this method CybersecurityServices.setDefaultSecurityNeedsLevels

Reply:
yes, element seems stored after call to CybersecurityServices.setDefaultSecurityNeedsLevels.